File: frameworks/web-framework-python/web_framework_benchmark/executors/app/Baseline11.py
from typing import Tuple

from web_framework_api import StatelessExecutor, HttpRequest


class Baseline11(StatelessExecutor):
    @staticmethod
    def _parse_values(request: HttpRequest) -> Tuple[int, int]:
        query_parameters = request.get_query_parameters()

        return int(query_parameters["a"]), int(query_parameters["b"])

    def do_get(self, request, response):
        a, b = Baseline11._parse_values(request)

        response.set_body(f"{a + b}")

    def do_post(self, request, response):
        a, b = Baseline11._parse_values(request)
        headers = request.get_headers()

        if "Transfer-Encoding" in headers:
            c = int(request.get_chunks()[0])
        else:
            c = int(request.get_body())

        response.set_body(f"{a + b + c}")

File: frameworks/web-framework-csharp/web_framework_benchmark/Executors/Baseline11.cs
namespace Executors;

using Framework;
using System.Text;

public class Baseline11 : StatelessExecutor
{
	private static void ParseValues(HttpRequest request, out int a, out int b)
	{
		IDictionary<string, string> queryParameters = request.GetQueryParameters();

		a = int.Parse(queryParameters["a"]);
		b = int.Parse(queryParameters["b"]);
	}

	public override void DoGet(HttpRequest request, HttpResponse response)
	{
		ParseValues(request, out int a, out int b);

		response.SetBody($"{a + b}");
	}

	public override void DoPost(HttpRequest request, HttpResponse response)
	{
		IDictionary<string, string> headers = request.GetHeaders();
		int c;

		ParseValues(request, out int a, out int b);

		if (headers.ContainsKey("Transfer-Encoding"))
		{
			c = int.Parse(Encoding.ASCII.GetString(request.GetChunks()[0]));
		}
		else
		{
			c = int.Parse(request.GetHttpBody());
		}

		response.SetBody($"{a + b + c}");
	}
}
